osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -1271,6 +1265,38 @@ protected override bool OnDragStart(DragStartEvent e)
 
                 return base.OnDragStart(e);
             }
+
+            protected override ScrollbarContainer CreateScrollbar(Direction direction)
+            {
+                return new PaddedScrollbar();
+            }
+
+            protected partial class PaddedScrollbar : OsuScrollbar
+            {
+                public PaddedScrollbar()
+                    : base(Direction.Vertical)
+                {
+                }
+            }
+
+            private const float top_padding = 10;
+            private const float bottom_padding = 80;
+
+            protected override float ToScrollbarPosition(float scrollPosition)
+            {
+                if (Precision.AlmostEquals(0, ScrollableExtent))
+                    return 0;
+
+                return top_padding + (ScrollbarMovementExtent - bottom_padding) * (scrollPosition / ScrollableExtent);
```

Comment:
I kinda question this math, namely the inner term of `ScrollbarMovementExtent - bottom_padding`. Why is that that, rather than `ScrollbarMovementExtent - (top_padding + bottom_padding)`? If `ScrollbarMovementExtent` is the true extent of movement across the Y axis, then why would it have only the _bottom_ padding subtracted?

In practice I think this may make the bottom padding 10 units smaller than possibly intended?

Reply:
Yes I likely got the math wrong, didn't check it thoroughly because I couldn't notice a visual issue.

Reply:
With your change alone the alignment doesn't match when click-dragging, and I'm not sure how to fix it 😅.

Reply:
I mean the full change would have to pretty clearly be applied to the functions in both directions, i.e.

```diff
diff --git a/osu.Game/Screens/Select/BeatmapCarousel.cs b/osu.Game/Screens/Select/BeatmapCarousel.cs
index ba20031509..aa7a8ba35c 100644
--- a/osu.Game/Screens/Select/BeatmapCarousel.cs
+++ b/osu.Game/Screens/Select/BeatmapCarousel.cs
@@ -1287,7 +1287,7 @@ protected override float ToScrollbarPosition(float scrollPosition)
                 if (Precision.AlmostEquals(0, ScrollableExtent))
                     return 0;
 
-                return top_padding + (ScrollbarMovementExtent - bottom_padding) * (scrollPosition / ScrollableExtent);
+                return top_padding + (ScrollbarMovementExtent - (top_padding + bottom_padding)) * (scrollPosition / ScrollableExtent);
             }
 
             protected override float FromScrollbarPosition(float scrollbarPosition)
@@ -1295,7 +1295,7 @@ protected override float FromScrollbarPosition(float scrollbarPosition)
                 if (Precision.AlmostEquals(0, ScrollbarMovementExtent))
                     return 0;
 
-                return ScrollableExtent * ((scrollbarPosition - top_padding) / (ScrollbarMovementExtent - bottom_padding));
+                return ScrollableExtent * ((scrollbarPosition - top_padding) / (ScrollbarMovementExtent - (top_padding + bottom_padding)));
             }
         }
 

```

but I'm not sure what issue I'm supposed to be seeing? What's the "alignment that doesn't match"?

Reply:
That seems to work. I've reduced the bottom padding as I originally visually aligned based on previous logic.